### PR TITLE
ipn/ipnlocal/peerapi: add endpoint to list local interfaces

### PR DESCRIPTION
**Pull request by maisem**
Signed-off-by: Maisem Ali <maisem@tailscale.com>

### CODE REVIEW

**Review by bradfitz — COMMENTED** [review]
routing table too while you're here?

**Comment by maisem** [comment]
> routing table too while you're here?

yep, just realized that I want that and the default route

**Comment by maisem** [comment]
> routing table too while you're here?

I'll do that in a different PR as it requires changing a bunch of code to get an OS independent view of the routing table

File: ipn/ipnlocal/peerapi.go [hunk 1]
```diff
@@ -581,6 +584,40 @@ This is my Tailscale device. Your device is %v.
...
+	if err != nil {
+		http.Error(w, err.Error(), 500)
+	}
+	w.Header().Set("Content-Type", "text/html; charset=utf-8")
+	fmt.Fprintln(w, "<h1>Interfaces</h1>")
+	fmt.Fprintf(w, "<h3>Default route is %q(%d)</h3>\n", dr.InterfaceName, dr.InterfaceIndex)
+
+	fmt.Fprintln(w, "<table>")
+	fmt.Fprint(w, "<tr>")
+	for _, v := range []any{"Index", "Name", "MTU", "Flags", "Addrs"} {
```
Comment (bradfitz) [inline ipn/ipnlocal/peerapi.go:607]: are we at the point where we're using `any` because its fewer characters than `string`? :)

**Review by bradfitz — APPROVED** [review]

File: ipn/ipnlocal/peerapi.go (hunk 1 above)
Comment (maisem) [reply ipn/ipnlocal/peerapi.go:607]: I had non-string values in the slice below, and this was added later. :)

**Pull request merged**